### ISSUE TITLE
AttributeError: 'OpenAIEmbeddings' object has no attribute 'deployment'

### ISSUE THREAD

**Issue by gullerg** [langchain-ai/langchain#3251]
get this error AttributeError: 'OpenAIEmbeddings' object has no attribute 'deployment' when deploying LangChain to DigitalOcean - however I don't get it locally. 

It seems rather odd, as when going through the source code, OpenAIEmbeddings indeed seem to have a 'deployment' attribute. What could cause this?

**Comment by MariuszT** [comment]
In my experience, the problem has occurred since version 0.0.145. I do not know the cause. You can do a downgrade to version 0.0.144.

**Comment by gullerg** [comment]
Getting this error after downgrading: TypeError: IndexFlat.search() missing 3 required positional arguments: 'k', 'distances', and 'labels'

Seems to be a regression issue

**Comment by AliMirlou** [comment]
@gullerg
The problem arises from [this PR](https://github.com/hwchase17/langchain/pull/3076) (specifically [this line](https://github.com/hwchase17/langchain/pull/3076/files#diff-d7a6d6752d04755e0881f71a73e707932889769c486afac9ede77b92ea125cbbR106))

The `deployment` has a default value though, so if you create a new object from it, it should have no problems.

Unless, you have saved the object created with an older version of LangChain in a file or something, and loaded it back with a newer version. That way, the `deployment` field does not even exist and you get this error.

**Comment by ademidun** [comment]
> Getting this error after downgrading: TypeError: IndexFlat.search() missing 3 required positional arguments: 'k', 'distances', and 'labels'
> 
> Seems to be a regression issue

I'm guessing the fix is downgrading to `0.0.144`?

**Comment by Laisky** [comment]
> @gullerg The problem arises from [this PR](https://github.com/hwchase17/langchain/pull/3076) (specifically [this line](https://github.com/hwchase17/langchain/pull/3076/files#diff-d7a6d6752d04755e0881f71a73e707932889769c486afac9ede77b92ea125cbbR106))
> 
> The `deployment` has a default value though, so if you create a new object from it, it should have no problems.
> 
> Unless, you have saved the object created with an older version of LangChain in a file or something, and loaded it back with a newer version. That way, the `deployment` field does not even exist and you get this error.

Thank you for the explanation. But may I ask if there is any way to update the old version files without re-run all these embeddings process?

**Comment by Laisky** [comment]
> > Getting this error after downgrading: TypeError: IndexFlat.search() missing 3 required positional arguments: 'k', 'distances', and 'labels'
> > Seems to be a regression issue
> 
> I'm guessing the fix is downgrading to `0.0.144`?

I solved this issue in my fork.

1. uninstall langchain `pip uninstall -y langchain`
2. reinstall langchain `pip install git+https://github.com/Laisky/langchain.git@c5505ed3`

3. add an new environment `OPENAI_EMBEDDINGS_DEPLOYMENT`, this is your azure deployment name of embeddings model.

```py
# -------------------------------------
# azure
# -------------------------------------
os.environ["OPENAI_API_TYPE"] = "azure"
os.environ["OPENAI_API_KEY"] = settings.OPENAI_AZURE_TOKEN
os.environ["OPENAI_API_BASE"] = settings.OPENAI_AZURE_API
os.environ["OPENAI_API_VERSION"] = "2023-05-15"
os.environ["OPENAI_EMBEDDINGS_DEPLOYMENT"] = "embedding"
# -------------------------------------
```

done
